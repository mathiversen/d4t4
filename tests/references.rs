use data_parser::{parse, Result};
use indoc::indoc;
use insta::assert_json_snapshot;

#[test]
fn it_can_parse_reference() -> Result<()> {
    let markup = indoc!(
        r#"{
            "value": "10px",
            "border": "border-width: ${value}"
        }"#
    );
    let x = parse(markup)?;
    assert_json_snapshot!(x);
    Ok(())
}

#[test]
fn it_can_parse_multiple_references() -> Result<()> {
    let markup = indoc!(
        r#"{
            "values": {
                "r": "10",
                "g": "10",
                "b": "100"
            },
            "two": "rgba(${values.r}, ${values.g}, ${values.b}, 0.1)",
            "tree": "${values.r}px"
        }"#
    );
    let x = parse(markup)?;
    assert_json_snapshot!(x);
    Ok(())
}
