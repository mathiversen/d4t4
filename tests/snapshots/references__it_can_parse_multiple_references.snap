---
source: tests/references.rs
expression: x
---
{
  "b": "100",
  "color": "rgba(10, 10, 100, 0.1)",
  "g": "10",
  "r": "10"
}
